kpet/data.py

```diff
@@ -387,26 +388,105 @@ def validate_host_type_regex(self):
                                   host_type_regex.pattern,
                                   host_types)
 
+    def convert_tree_arches(self):
+        """
+        Get a list of available arches for each tree
+        Returns:
+            Raises a schema.Invalid exception when finding an invalid regex
+        """
+
+        trees = {}
+        # Create a dict {tree: arches,} where arches is a list of regexes
+        for tree in self.trees:
+            trees[tree] = []
+            if "arches" not in self.trees[tree]:
+                self.trees[tree]["arches"] = [".*"]
+
+            for arch_regex in self.trees[tree]["arches"]:
+                temp_trees = []
+                arch_regex = re.compile(arch_regex)
+                for arch in self.arches:
+                    if arch_regex.fullmatch(arch):
+                        temp_trees.append(arch)
+                if temp_trees == []:
+                    error = ("One of 'trees: arches:' regexes\n" +
+                             "doesn't match any of the available arches\n" +
+                             "The regex: '{0}'\n" +
+                             "The avaliable arches: {1}")
+                    raise Invalid(error, arch_regex.pattern,
+                                  self.arches)
+                trees[tree] += temp_trees.copy()
+
+            trees[tree] = list(dict.fromkeys(trees[tree]))  # remove duplicity
+            self.trees[tree]["arches"] = trees[tree]
```

Comment:
Switching to sets would let you replace the above two lines with this:

    self.trees[tree]["arches"] = list(tree_arches)